src/services/ConsultationDialogueService.js

```diff
@@ -0,0 +1,41 @@
+const Dialog = require("../models/Dialog");
+
+function parserPageArgs(property, value) {
+  const passed = parseInt(value, 10);
+  if (property === "limit") {
+    if (Number.isNaN(passed) || !passed) return 10;
+  } else if (Number.isNaN(passed) || !passed) return 1;
+  return passed;
+}
+
+module.exports = async (req, res) => {
+  const { limit, page } = req.headers;
+
+  const options = {
+    limit: parserPageArgs("limit", limit),
+    page: parserPageArgs("page", page),
+  };
+  const dialogues = await Dialog.find(req.query, null, {
+    limit: options.limit,
+    skip: (options.page - 1) * options.limit,
+  }).catch((error) =>
```

Comment:
Don't use `then().catch()` anymore, use **try catch**. Thus:
```javascript
try {
  const result = await request();
  
} catch (err) {
  // If it fails it will automatically fall here
}
`